### PR TITLE
Smooth labels

### LINKED ISSUE #245: [Bug] relabel set to `true` in `opening_labels` while it should be `false`
https://github.com/clEsperanto/CLIc/blob/efcd8fa05d0eddeb297460bb9471a782a4c59a9f/clic/src/tier7.cpp#L176-L182

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hi @StRigaud ,

this adds `smooth_labels()` and fixes a bug in opening_labels.

closes #245 
closes https://github.com/clEsperanto/pyclesperanto/issues/144

Let me know what you think!

Cheers,
Robert

### CODE REVIEW

**Comment by StRigaud** [comment]
But of course, It is amazing @haesleinhuepf !

File: tests/tier8/test_smooth_labels.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                                                  2, 2, 2, 2, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 2, 2,
+                                                  2, 2, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 2, 2, 2, 2 };
+  std::array<uint32_t, 8 * 9 * 1>       output;
+};
+
+TEST_P(TestSmoothLabels, execute)
+{
+  std::string param = GetParam();
+  cle::BackendManager::getInstance().setBackend(param);
+  auto device = cle::BackendManager::getInstance().getBackend().getDevice("RTX", "all");
```
Comment (haesleinhuepf) [inline tests/tier8/test_smooth_labels.cpp:24]:
```suggestion
  auto device = cle::BackendManager::getInstance().getBackend().getDevice("", "all");
```

**Comment by StRigaud** [comment]
it should be good now :)

**Review by StRigaud — APPROVED** [review]

**Comment by kevinyamauchi** [comment]
This would be super useful for me! Thanks @haesleinhuepf and @StRigaud !

**Comment by StRigaud** [comment]
> This would be super useful for me! Thanks @haesleinhuepf and @StRigaud !

@kevinyamauchi, don't hesitate to make some feature or filter request 😉

**Comment by kevinyamauchi** [comment]
> @kevinyamauchi, don't hesitate to make some feature or filter request 😉

@StRigaud , be careful what you wish for 😉

**Pull request merged**